src/gsAssembler/gsQuadrature.h

```diff
@@ -0,0 +1,82 @@
+/** @file gsQuadrature.h
+
+    @brief Creates a variety of quadrature rules
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): A. Mantzaflaris
+*/
+
+#pragma once
+
+#include <gsAssembler/gsGaussRule.h>
+#include <gsAssembler/gsLobattoRule.h>
+
+namespace gismo
+{
+
+/// Helper class for obtaining a quadrature rule
+struct gsQuadrature
+{
+    /// Quadrature rule types
+    enum rule
+    {
+        GaussLegendre = 1, ///< Gauss-Legendre quadrature
+        GaussLobatto  = 2  ///< Gauss-Lobatto quadrature
+    };
+
+    /// Constructs a quadrature rule based on input \a options
+    template<class T>
+    static gsQuadRule<T> get(const gsBasis<T> & basis,
+                             const gsOptionList & options, int fixDir = -1)
+    {
+        const int qu = options.askInt("quRule", GaussLegendre);
+        const T       quA = options.getReal("quA");
+        const index_t quB = options.getInt ("quB");
+        const gsVector<index_t> nnodes = numNodes(basis,quA,quB,fixDir);
+        return get<T>(qu, nnodes);
+    }
+
+    /// Constructs a quadrature rule based on input \a options
+    template<class T>
+    static inline gsQuadRule<T> get(int qu, gsVector<index_t> const & numNodes)
```

Comment:
why "int"? shouln't it be "rule"?

Reply:
GISMO_ERROR will be invoked for invalid values, so passing int 1 or 2 is perfectly fine, and I want ot allow it, as it is the same as the enum

Reply:
This seems to be a general question. Doing the same, e.g., for InterfaceStrategy and BoundaryStrategy in assemblers and dof mapper etc. would also save many casts...

Reply:
yes, in fact in the assemblers these will go away completely, they are in the options instead..